opentelemetry-etw-metrics/src/exporter/mod.rs

```diff
@@ -0,0 +1,93 @@
+use opentelemetry::{
+    global,
+    metrics::{MetricsError, Result},
+};
+use opentelemetry_proto::tonic::collector::metrics::v1::ExportMetricsServiceRequest;
+use opentelemetry_sdk::metrics::{
+    data::{ResourceMetrics, Temporality},
+    exporter::PushMetricsExporter,
+    reader::{AggregationSelector, DefaultAggregationSelector, TemporalitySelector},
+    Aggregation, InstrumentKind,
+};
+
+use async_trait::async_trait;
+use prost::Message;
+
+use std::fmt::{Debug, Formatter};
+
+use crate::etw;
+
+pub struct MetricsExporter {}
+
+impl MetricsExporter {
+    pub fn new() -> MetricsExporter {
+        etw::register();
+
+        MetricsExporter {}
+    }
+}
+
+impl Default for MetricsExporter {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TemporalitySelector for MetricsExporter {
+    fn temporality(&self, kind: InstrumentKind) -> Temporality {
+        match kind {
+            InstrumentKind::Counter
+            | InstrumentKind::ObservableCounter
+            | InstrumentKind::ObservableGauge
+            | InstrumentKind::Histogram
+            | InstrumentKind::Gauge => Temporality::Delta,
+            InstrumentKind::UpDownCounter | InstrumentKind::ObservableUpDownCounter => {
+                Temporality::Cumulative
+            }
+        }
+    }
+}
+
+impl AggregationSelector for MetricsExporter {
+    fn aggregation(&self, kind: InstrumentKind) -> Aggregation {
+        DefaultAggregationSelector::new().aggregation(kind)
+    }
+}
+
+impl Debug for MetricsExporter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str("ETW metrics exporter")
+    }
+}
+
+#[async_trait]
+impl PushMetricsExporter for MetricsExporter {
+    async fn export(&self, metrics: &mut ResourceMetrics) -> Result<()> {
+        let proto_message: ExportMetricsServiceRequest = (&*metrics).into();
+
+        let mut byte_array = Vec::new();
+        proto_message
+            .encode(&mut byte_array)
+            .map_err(|err| MetricsError::Other(err.to_string()))?;
+
+        let result = etw::write(&byte_array);
```

Comment:
https://github.com/open-telemetry/opentelemetry-rust-contrib/blob/main/opentelemetry-user-events-metrics/src/tracepoint/mod.rs#L43
We can validate the part here, right? If the payload is exceeding ~64kb, drop it, and log via global error handler.

Reply:
`tracelogging::write_event` will return a non-zero error code which can be logged by callers of `etw::write`. Should `etw::write` log the global error too?

Reply:
okay the current code seem to be logging errors already, that is good enough now.